### DISCUSSION TITLE
STDOUT pipe/redirect doc page: I do not understand how to use it

### DISCUSSION THREAD

**Discussion by aborruso** [saulpw/visidata#1106] [Q&A] [answered]
Hi,
I have read [this page](https://www.visidata.org/docs/pipes/). 

If I run one of the examples - `ps -ef | vd | tail -n +2` - I do not have the output in stdout, but I have it in vd.

![image](https://user-images.githubusercontent.com/30607/132179156-ec5ecb69-fa6f-42cb-a435-42e8736ab36f.png)

In that page I have

> Use `vd -o-` to send directly to the terminal when not redirecting stdout (it's not necessary if already redirected).

If I run `vd -o- input.csv | head` once again I have it in vd and not in stdout.

![image](https://user-images.githubusercontent.com/30607/132179384-e3b427cd-865a-45db-852b-c23ccbebdd80.png)

The same if I run `vd input.csv | head`.

Probably I'm stupid and I did not understand this docs page. 

Thank you

**Comment by saulpw** [comment, marked as answer]
@aborruso  The pipeline will only continue after VisiData exits.  And did you use `Ctrl+Q` to exit VisiData?  Otherwise there will be no top sheet, and no data will be output.

**Comment by aborruso** [comment]
@saulpw thank you. 

A last (probably stupid question). I do not understand `vd -o-` use case. When is it useful to use it? What purpose does it serve?

**Reply by saulpw** [reply]
For clarity in a script, maybe; or if you wanted the output to go to the literal terminal without needing to pipe to 'cat'.  Maybe not essential, but it's a standard CLI idiom and rounds out the interface.

Also, this option works in batch mode too, which doesn't require Ctrl+Q.  If you replay a script, or just as a converter (scriptless batch mode), then the final sheet will be output to the -o/--output option, if there is one (or the output terminal is redirected). So `vd -b foo.jsonl -o - --save-filetype tsv` will dump the jsonl file as a tsv so you could see it in the terminal.

**Comment by hoclun-rigsep** [comment]
`vd -o-` complains of "setting unknown option" and does not produce the intended behavior. `vd -o -` works as expected. v2.7.1 and python 3.9.2.
